configs/config.go

```diff
@@ -0,0 +1,186 @@
+	Poller           PollerConfig           `mapstructure:"poller"`
+	Committer        CommitterConfig        `mapstructure:"committer"`
+	FailureRecoverer FailureRecovererConfig `mapstructure:"failure_recoverer"`
+	Storage          StorageConfig          `mapstructure:"storage"`
+}
+
+func LoadConfig(cfgFile string) error {
+	if cfgFile != "" {
+		viper.SetConfigFile(cfgFile)
+		if err := viper.ReadInConfig(); err != nil {
+			return fmt.Errorf("error reading config file, %s", err)
+		}
+	} else {
+		viper.SetConfigName("config")
+		viper.AddConfigPath("./configs")
+
+		if err := viper.ReadInConfig(); err != nil {
+			return fmt.Errorf("error reading config file, %s", err)
+		}
+
+		viper.SetConfigName("secrets")
+		err := viper.MergeInConfig()
+		if err != nil {
+			return fmt.Errorf("error loading secrets file: %v", err)
+		}
+	}
+
+	// sets e.g. RPC_URL to rpc.url
+	replacer := strings.NewReplacer(".", "_")
+	viper.SetEnvKeyReplacer(replacer)
+
+	viper.AutomaticEnv()
+
+	// viper.BindPFlag("rpc.url", rootCmd.PersistentFlags().Lookup("password"))
+
+	return nil
+}
+
+func GetRPCConfig() RPCConfig {
+	return RPCConfig{
+		URL: viper.GetString("rpc.url"),
+	}
+}
+
+func GetLogConfig() LogConfig {
+	return LogConfig{
+		Level:  viper.GetString("log.level"),
+		Pretty: viper.GetBool("log.pretty"),
+	}
+}
+
+func GetPollerConfig() PollerConfig {
+	return PollerConfig{
+		Enabled:       viper.GetBool("poller.enabled"),
+		Interval:      viper.GetInt("poller.interval"),
+		BlocksPerPoll: viper.GetInt("poller.blocks_per_poll"),
+		FromBlock:     viper.GetInt("poller.from_block"),
+		UntilBlock:    viper.GetInt("poller.until_block"),
+	}
+}
+
+func GetCommitterConfig() CommitterConfig {
+	return CommitterConfig{
+		Enabled:   viper.GetBool("committer.enabled"),
+		Interval:  viper.GetInt("committer.interval"),
+		BatchSize: viper.GetInt("committer.batch_size"),
+	}
+}
+
+func GetFailureRecovererConfig() FailureRecovererConfig {
+	return FailureRecovererConfig{
+		Enabled:   viper.GetBool("failure_recoverer.enabled"),
+		Interval:  viper.GetInt("failure_recoverer.interval"),
+		BatchSize: viper.GetInt("failure_recoverer.batch_size"),
+	}
+}
+
+func GetStorageConfig() StorageConfig {
+	return StorageConfig{
+		Staging:      getStorageConnectionConfig(StorageTypeStaging, StorageDriverClickhouse),
+		Main:         getStorageConnectionConfig(StorageTypeMain, StorageDriverClickhouse),
+		Orchestrator: getStorageConnectionConfig(StorageTypeOrchestrator, StorageDriverMemory),
+	}
+}
+
+func getStorageConnectionConfig(storageType StorageType, driver StorageDriver) StorageConnectionConfig {
+	return StorageConnectionConfig{
+		Driver:     driver,
+		Clickhouse: getClickhouseConfig(storageType),
+		Memory:     getMemoryConfig(storageType),
+	}
+}
+
+func getClickhouseConfig(storageType StorageType) ClickhouseConfig {
+	return ClickhouseConfig{
+		Host:     viper.GetString(fmt.Sprintf("storage.%s.clickhouse.host", storageType)),
```

Comment:
what does GetString return if the string doesn't exist? normal Go flow returns err in such cases

Reply:
Default value, so for string it's an empty string